common/utils.py

```diff
@@ -222,40 +250,15 @@ def run_in_subprocess(func: functools.partial, ttl: int) -> Any:
         Any: The value returned by 'func'
     """
 
-    def wrapped_func(func: functools.partial, queue: multiprocessing.Queue):
-        try:
-            print(f"In subprocess")
-            result = func()
-            print("Putting result into queue")
-            queue.put(result)
-            print("Put thingy into the queue")
-        except (Exception, BaseException) as e:
-            # Catch exceptions here to add them to the queue.
-            queue.put(e)
-
-    # Use "fork" (the default on all POSIX except macOS), because pickling doesn't seem
-    # to work on "spawn".
-    ctx = multiprocessing.get_context("fork")
-    queue = ctx.Queue()
-    process = ctx.Process(target=wrapped_func, args=[func, queue])
-
-    process.start()
-
-    process.join(timeout=ttl)
-
-    if process.is_alive():
-        process.terminate()
-        process.join()
-        raise TimeoutError(f"Failed to {func.func.__name__} after {ttl} seconds")
-
-    # Raises an error if the queue is empty. This is fine. It means our subprocess timed out.
-    print("Pulling from the queue")
-    result = queue.get(block=False)
-
-    # If we put an exception on the queue then raise instead of returning.
-    if isinstance(result, Exception):
-        raise result
-    if isinstance(result, BaseException):
-        raise Exception(f"BaseException raised in subprocess: {str(result)}")
-
-    return result
+    executor = concurrent.futures.ThreadPoolExecutor(max_workers=1)
+
+    try:
+        future = executor.submit(func)
+        return future.result(timeout=ttl)
+    except concurrent.futures.TimeoutError as e:
+        bt.logging.error(f"Failed to complete '{name}' within {ttl} seconds.")
+        raise TimeoutError(f"Failed to complete '{name}' within {ttl} seconds.") from e
+    finally:
```

Comment:
do we not want to also log other exceptions / base exceptions if they occur?

Reply:
They'll be raised and the caller should catch/log them